jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java

```diff
@@ -308,6 +311,19 @@ else if (tokenstart >= 0)
                                     quoted = false;
                                     continue;
                                 }
+
+                                if (_compliance == CookieCompliance.RFC6265)
+                                {
+                                    // Rejected cookie-octet characters
+                                    // US-ASCII characters excluding CTLs,
+                                    // whitespace DQUOTE, comma, semicolon,
+                                    // and backslash
+                                    if (Character.isISOControl(c) || Character.isWhitespace(c) || c == '\\')
+                                    {
+                                        reject = true;
```

Comment:
I don't like this.  Once we have seen an illegal character, I do not think it is safe to continue parsing and discovering other cookies after a ';' - which may not be a real ';' but something injected by an attacker who used the invalid characters to get past some client restrictions.     If we are being RFC6265 compliant, then once we see a bad character, we should reject the entire header... problably entire request with Bad Request!

Reply:
If we are being RFC6265 compliant then we should parse using the rules it lays out.

1. Split by ';' first. (this means DQUOTE pairs are subservient to the `;`.  the `;` can never be in a DQUOTE).
2. Parse each section on it's own.
3. Discard / Reject each section that fails it's rules.

Reply:
See the issue for details ... #3985

Reply:
I don't see anywhere in RFC6265 where it specifies that alogithm for the Cookie header.  It definitely does not say anything about rejecting or discarding malformed cookies and continuing to search for other ones.

However, we can be forgiving in some circumstances and the case from #3985 is OK:
```
Cookie: testcookie; lmrelayserver.id=abcd; lmrelayserver.detail=efg
```
There are no special characters involved and we can be sure that the ; really is a ;

However, I'm a bit more cautious when it comes to actual illegal characters in the value.  We may well be protected by HTTP/1 and HTTP/2 RFCs not allowing special characters in the header value... but perhaps not.  My concern is that once we start seeing non legal characters in a Cookie value, then we can no longer trust that a following ; really is a ; and not just a tricked encode UTF8 character that has somehow got past a bad client.

So I'm +1 for ignoring missing = and continuing on, but I'm -1 on continuing on after illegal characters.  My +0 compromise is that we can continue on if the illegal character is not a control character.

Reply:
As for the algorithm - https://tools.ietf.org/html/rfc6265#section-5.2

Yes, I know that's for the User-Agent side.
But if the User-Agent is rejecting cookies for being bad based on that algorithm, we need to reject for the same reasons.
Otherwise there's no point in being lenient.

Reply:
The algorithm in CookieCutter appears to allow DQOUTED cookie-name, that's also a reject.
Only the cookie-value is allowed to be DQUOTED.

Reply:
As I said, I'm fine with rejecting cookies based on bad formatting, but not on bad characters.
If there is an iso control character in the cookie, then that has to be a hard failure and not a reject and continue.

Reply:
"in the cookie" is at the heart of this.

A Cookie header can have more than 1 cookie.

To me, `foo; bar="val; baz.b=123"; z=x` is 4 cookies.

Num | Cookie | Parsed name=value
--- | ----- | -------
1 | `foo` | (rejected, lacking `=` sign)
2 | `bar="val` | (rejected, lacking matching DQUOTE)
3 | `baz.b=123"` | (rejected, lacking matching DQUOTE)
4 | `z=x` | name=`z`, value=`x`

Reply:
Joakime, I agree that a strict interpretation of RFC6265 would have the result you describe above (or perhaps even rejecting all cookies from that header - or perhaps rejecting the request with a 400).

However, if we consider degrees of leniency, we have to consider the norms established by RFC2616 for header value parameters.  Typically you see ABNF like:
```
       something      = token *( ";" parameter )
       parameter      = attribute "=" value
       attribute      = token
       value          = token | quoted-string
       quoted-string  = ( <"> *(qdtext | quoted-pair ) <"> )
       qdtext         = <any TEXT except <">>
``` 
So there is plenty of code out there that will happily quote a `;` - and I definitely recall being asked to allow such leniant cookies and we have a lenient test case for:
```java
            Arguments.of("foo=\"bar;baz\"", "foo", "bar;baz"),
```

So I really think we don't want to get too radical here.  Let's step back again!  The problem at hand is dealing with simple misformed cookie headers like:
```
Cookie: testcookie; lmrelayserver.id=abcd; lmrelayserver.detail=efg
```
we can do that without making significant changes to our other leniency cases nor without giving extra leniency to special characters.    I think making a small change for just that leniency case is reasonable - but let's not make more problems for ourselves by changing existing accepted cookies or being too lenient

Reply:
If the browser handles parsing / rejecting in a different way then us, isn't that a bug on our part?

Reply:
No - because a set-cookie and a cookie header are completely different syntax that have an entirely different meaning for ";"!  With set-cookie, the ";" is used to separate out parameters for the cookie (max-age, path etc.) while for cookie header the ";" separates entirely different cookies (it should have been ",").

Whoever has written the cookie RFCs over the years has made a complete pigs ear of it and what we have today is a delicate pile of fragile comprimises between strict RFC adherence and leniency for past sins.    We simply cannot take the risk of suddenly in a point release changing the way we interpret this fragile collection of compromises!  Specially when there is absolutely no need to do so to fix #3985!  

All that we need to do to fix the issue at hand is to change how we handle a missing '='.  There is no need to completely change the compromises that we have for quotes etc.

Reply:
This has been moved to a `isRFC6265RejectedCharacter(char c)` method later in the class.
Only Control characters are tested for now.